File: Ferofly/app/src/main/java/com/ferofly/ferofly/ui/MainFragment.java
package com.ferofly.ferofly.ui;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.core.content.ContextCompat;
import androidx.core.content.res.ResourcesCompat;
import androidx.databinding.DataBindingUtil;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.ferofly.ferofly.R;
import com.ferofly.ferofly.databinding.FragmentMainBinding;
import com.shuhart.stepview.StepView;

import java.util.ArrayList;

public class MainFragment extends Fragment {

    FragmentMainBinding binding;
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = DataBindingUtil.inflate(inflater,R.layout.fragment_main,container,false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        binding.stepView.getState()
                .selectedTextColor(ContextCompat.getColor(getContext(), R.color.black))
                .animationType(StepView.ANIMATION_CIRCLE)
                .selectedCircleColor(ContextCompat.getColor(getContext(), R.color.black))
                .selectedCircleRadius(getResources().getDimensionPixelSize(R.dimen.cardview_compat_inset_shadow))
                .selectedStepNumberColor(ContextCompat.getColor(getContext(), R.color.colorprimary))
                // You should specify only stepsNumber or steps array of strings.
                // In case you specify both steps array is chosen.
                .steps(new ArrayList<String>() {{
                    add("Cooking");
                    add("Picked");
                    add("On way");
                    add("Delivered");
                    add("Done");

                }})
                // You should specify only steps number or steps array of strings.
                // In case you specify both steps array is chosen.
                .stepsNumber(4)
                .animationDuration(getResources().getInteger(android.R.integer.config_shortAnimTime))
                .stepLineWidth(getResources().getDimensionPixelSize(R.dimen.dp1))
                .textSize(getResources().getDimensionPixelSize(R.dimen.sp14))
                .stepNumberTextSize(getResources().getDimensionPixelSize(R.dimen.sp16))
                .typeface(ResourcesCompat.getFont(getContext(), R.font.calibri))
                // other state methods are equal to the corresponding xml attributes
                .commit();
        binding.trackYourOrder.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Uri gmmIntentUri = Uri.parse("geo:0,0?q=2nd Floor, Hno12,Road np 37, Back side of MGS hospital, West Punjabi Bagh, Pubjabi Bagh, Delhi, 110026, India");
                Intent mapIntent = new Intent(Intent.ACTION_VIEW, gmmIntentUri);
                mapIntent.setPackage("com.google.android.apps.maps");
                startActivity(mapIntent);
            }
        });

    }
}

File: MarketwinTask/app/src/main/java/com/example/marketwintask/ui/MainFragment.kt
package com.example.marketwintask.ui

import android.app.Activity
import android.app.Activity.RESULT_OK
import android.content.ContentResolver
import android.content.Intent
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.net.Uri
import android.os.Bundle
import android.provider.MediaStore
import android.service.media.MediaBrowserService
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import com.example.marketwintask.R
import com.example.marketwintask.databinding.FragmentMainBinding
import java.io.FileNotFoundException
import java.io.InputStream


class MainFragment : Fragment() {

    private lateinit var binding: FragmentMainBinding
    private var imagereq = 21
    private lateinit var bitmap: Bitmap
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = DataBindingUtil.inflate(inflater,R.layout.fragment_main,container,false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding.selectImgBtn.setOnClickListener{


            val i = Intent()
            i.type = "image/*"
            i.setAction(Intent.ACTION_GET_CONTENT)
            startActivityForResult(i , imagereq)
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        if (requestCode == imagereq && resultCode == RESULT_OK) {
            bitmap= data?.extras?.get("image/*") as Bitmap
            binding.mySelectedImg.setImageBitmap(bitmap)
        }
    }
}